add a table in marimo to summarize selected corpora

tanhaow opened the pull request:
**Associated Issue(s):** resolves #299 

- [ ] enable directory selection 

### Changes in this PR
- Created a summary table in Marimo to display information of selected original corpora (`filename`, `total lines`, `body text lines`, `footnote lines`, `last updated`).
- Created two layouts for Ben and Ed to decide which is preferred:

1. **Side-by-side view** (branch: `feature/marimo-ui-for-multi-orginal-corpora`):
<img width="2320" height="1578" alt="image" src="https://github.com/user-attachments/assets/e94209f4-4bb2-4b6c-9c9f-41f7959150b7" />

2. **Sequential view** (branch: `feature/marimo-ui-multi-original-sequential`):
<img width="1568" height="1940" alt="image" src="https://github.com/user-attachments/assets/8f53e5a1-a0e7-473b-9665-9b875eefc0b2" />

### Reviewer Checklist
- [ ] Review the layout and suggest enhancements

rlskoeser commented on src/remarx/app/quote_finder.py:
```diff
@@ -155,6 +158,70 @@ def _(mo, original_csv_browser, reuse_csv_browser):
+            "total_lines": total,
+            "body_text_lines": body,
+            "footnote_lines": foot,
+            "last_updated": last_updated,
```
Since these are meant for human consumption, I suggest leaving the `_` out to make it more readable.

tanhaow commented:
@rlskoeser Thanks for all the great suggestions. This PR is ready for review again. Changes include:
- added default quote finder out logic (`remarx-data/quote-finder-output`), and refactored corresponding ui.
- have summary information for both sets of selected corpora, and left all the `_` out in the header.
- support directory selection for original corpora --> I cannot an easy way to support this, but marimo file browser widget has a "select all" feature, which supports select all files in the directory.

rlskoeser commented:
Some comments from testing the app (not strictly related to this PR):

In corpus builder:
-  Help text for default paths seems overly verbose (I didn't notice on the other PR); I can suggest some edits, maybe a separate PR (or maybe we solicit input from the project team to revise)
- Output folder is not updating when I change the toggle for original / reuse output (both reported in the app and actual)

rlskoeser commented on src/remarx/app/utils.py:
```diff
@@ -127,6 +137,57 @@ def create_temp_input(
+        "corpus": path.name,
+        "total lines": int(total_sentences),
+        "body text lines": body_sentences,
+        "footnote lines": footnote_sentences,
```
Revise labels to more precise & consistent. 
```suggestion
        "filename": path.name,
        "total sentences": int(total_sentences),
        "body text sentences": body_sentences,
        "footnote sentences": footnote_sentences,
```

rlskoeser commented on CHANGELOG.md:
```diff
@@ -19,6 +19,10 @@
 
+### Application
+
+- Add a summary table in `Quote Finder` notebook displaying user-selected original corpora's information
```
Now applies to both corpora

```suggestion
- Add summary tables in `Quote Finder` to display details for selected input corpora
```
